### ISSUE TITLE
Spike: Robot discovery/connection engineering documentation

### ISSUE THREAD

**Issue by mcous** [Opentrons/opentrons#4778]
## overview

This ticket is a research spike to ensure we have a clear, shared understanding of the technical mechanisms and processes that are in place so that a given robot can be "discovered" by the app during its first and subsequent connections

This documentation should be fairly visual, because it will be an important input into an FMEA of the system

## acceptance criteria

- [x] Description and diagrams of wired link-local discovery
- [x] ~Description and diagrams of wireless DHCP discovery~ (Out of scope for now)
- [x] ~(Maybe?) Description and diagrams of discovery when both interfaces are in use~ (Out of scope for now)

Note: discuss after each of the checkboxes above are tackled

**Comment by mcous** [comment]
Documentation to unblock FMEA generated, available at: https://coda.io/d/Robot-Discovery-Research_dNGxVkumwhM (internal, Opentrons-access-only link)
